### PR TITLE
Fix flickering tests

### PR DESCRIPTION
**Pull request by pezholio**
## Trello card URL:

https://trello.com/c/8EMdqNfY/835-fix-flickering-failing-tests

## Changes in this PR:

Finally found the issue behind this. The fix is to remove all whitespace from the titles when using `sort_by`, which mimics the behaviour of Postgres's `ORDER BY`

### CODE REVIEW

File: spec/presenters/school_vacancies_presenter_spec.rb [hunk 1]
```diff
@@ -18,78 +18,86 @@
   end
   let!(:published_vacancies) { create_list(:vacancy, 7, :published, school: school) }
 
+  let(:vacancy_ids) do
+    vacancies.sort_by { |vacancy| vacancy['job_title'].delete(' ') }.reverse.pluck(:id)
+  end
+
```
Comment (erbridge) [inline spec/presenters/school_vacancies_presenter_spec.rb]: This is specifically for when it's sorted by `column: 'job_title', order: 'desc'`, and isn't general. I'm not sure it makes sense to pull out to here.

Comment (pezholio) [reply spec/presenters/school_vacancies_presenter_spec.rb]: Ah, well spotted. I just got refactor-happy there :D

Comment (pezholio) [reply spec/presenters/school_vacancies_presenter_spec.rb]: I've wrapped the sorting stuff in a `describe` block to be clearer about what we're doing here

**Comment by pezholio** [comment]
Ready for another go round now, I've also stopped using the `let!`s to set up state, as these get called for each spec, slowing the execution time down dramatically.

File: spec/presenters/school_vacancies_presenter_spec.rb [hunk 2]
```diff
@@ -1,25 +1,77 @@
...
+      3.times do
+        expired_vacancy = build(:vacancy, :expired, school: @school)
+        expired_vacancy.save(validate: false)
+        expired_vacancies << expired_vacancy
+      end
+      expired_vacancies
     end
-    expired_vacancies
+    @published_vacancies = create_list(:vacancy, 7, :published, school: @school)
   end
```
Comment (erbridge) [inline spec/presenters/school_vacancies_presenter_spec.rb]: :+1:

**Pull request merged**